### PR TITLE
Keep track of a publisher's status

### LINKED ISSUE #850: Publisher Status
Before we implement `status` on publishers, I want to make sure we get the states down correctly:

`created` as opposed to `create`
`onboarding` as opposed to `updating`
`active` 
`suspended`
`deleted` which is just `archived`

@nvonpentz let me know if you agree or disagree.

### PR DESCRIPTION
**Pull request by nvonpentz**
Resolves #850 

Note: This PR does nothing to address existing publishers.  Please see my [comment](https://github.com/brave-intl/publishers/issues/850#issuecomment-388955125) on the challenges and possible half solutions for that.

* Add new table PublisherStatusUpdates

* Automatically create a PublisherStatusUpdate (created) upon creation of a publisher

* Create a PublisherStatusUpdate (onboarding) upon verifying email

* Create a PublisherStatusUpdate (active) upon agreeing to TOS and addressing 2fa prompt

* Add new Publisher & PublisherStatusUpdate fixtures

* Add tests

Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave-intl/publishers/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Tagged reviewers.

Test Plan:

Reviewer Checklist:

Tests
- [ ] Adequate test coverage exists to prevent regressions

Security:
- [ ] No raw SQL -- Always prefer ActiveRecord query helpers ([more info on StackOverflow](https://stackoverflow.com/questions/41410752/rails-5-sql-injection#41452695))
- [ ] XSS is mitigated -- Avoid `html_safe` and `raw`; escape untrusted content from users and 3rd party APIs ([Rails XSS guide](https://brakemanpro.com/2017/09/08/cross-site-scripting-in-rails) and [OWASP XSS Prevention Cheat Sheet](https://www.owasp.org/index.php/XSS_(Cross_Site_Scripting)_Prevention_Cheat_Sheet))

### CODE REVIEW

File: app/models/publisher.rb [hunk 1]
```diff
@@ -198,7 +203,30 @@ def has_verified_channel?
...
   end
 
+  def last_status_update
+    status_updates.first
+  end
+
   private
+  
+  def set_created_status
+    created_publisher_status_update = PublisherStatusUpdate.new(publisher: self, status: "created")
```
Comment (yachtcaptain23) [inline app/models/publisher.rb]: Would you convert the statuses into constants? I recommend keep track of statuses in something similar to:

`STATUSES = [ONBOARDING, CREATED, ACTIVE, SUSPENDED]`

Comment (nvonpentz) [reply app/models/publisher.rb]: I added a list a constant `ALL_STATUSES` on the `PublisherStatusUpdates` model, and added a validation that the status must be in the list of all statuses.  Please let me know if you had something else in mind @yachtcaptain23.

File: app/models/publisher_status_update.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class PublisherStatusUpdate < ApplicationRecord
+  has_paper_trail
+  
+  ALL_STATUSES = ["created", "onboarding", "active"]
```
Comment (yachtcaptain23) [inline app/models/publisher_status_update.rb]: Would you add the `suspended` state to this list?

Additionally, let's make these constants:

```
CREATED = 'created'
ONBOARDING = 'onboarding'
ACTIVE = 'active'
SUSPENDED = 'suspended'
ALL_STATUSES = [CREATED, ONBOARDING, ACTIVE, SUSPENDED]
``

File: app/models/publisher_status_update.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class PublisherStatusUpdate < ApplicationRecord
+  has_paper_trail
```
Comment (yachtcaptain23) [inline app/models/publisher_status_update.rb]: Make sure you delete this line. This model essentially is its own paper_trail.

I realize it seems silly to run our own paper trail, but essentially we don't want to bloat 1 table too hard to affect the rest of the database which depends on a `versions` table.

**Pull request merged**